avocado/core/job.py

```diff
@@ -515,6 +527,9 @@ def run(self):
             if not settings.get_value('runner.behavior', 'keep_tmp_files',
                                       key_type=bool, default=False):
                 data_dir.clean_tmp_files()
+            if self.time_end == -1:
+                self.time_end = time.time()
+                self.time_elapsed = self.time_end - self.time_start
```

Comment:
The approach chosen here is: if user does nothing, all are calculated during run.  If user changes any of them, he's responsible for making sure the changes are consistent.  The selftest shows that approach.  Eventually, this may receive more documentation, as users of the Job API appear.